src/lib.rs

```diff
@@ -78,16 +78,10 @@ enum KeyBuffer<'a, const SIZE: usize> {
     Owned(Box<[u8; SIZE]>),
 }
 
-impl<'a, const SIZE: usize> From<&'a mut [u8; SIZE]> for KeyBuffer<'a, SIZE> {
-    fn from(buf: &'a mut [u8; SIZE]) -> Self {
-        Self::Borrowed(buf)
-    }
-}
-
-#[cfg(feature = "alloc")]
-impl<const SIZE: usize> From<Box<[u8; SIZE]>> for KeyBuffer<'static, SIZE> {
-    fn from(buf: Box<[u8; SIZE]>) -> Self {
-        Self::Owned(buf)
+impl<'a, const SIZE: usize> KeyBuffer<'a, SIZE> {
+    #[cfg(feature = "alloc")]
+    fn to_owned(&self) -> KeyBuffer<'static, SIZE> {
+        KeyBuffer::Owned(Box::new(*self.as_ref()))
```

Comment:
This line still copies the array on the stack right?
Using the `util:util::alloc_boxed_array` trick + copy from slice should prevent this

Reply:
Maybe centralizing the `util:alloc_boxed_array` into a `KeyBuffer::new_owned()` method is also better, since you need to use this multiple times below.

Reply:
I really wish there was a non-unsafe and stable way to do this with the standard library.. This is one of the terrible things about Rust, that something so simple is so hard. Yes. I'll try to fix that. I added a test for the thing you describe. But `keypair` in general needs so incredibly large stack, so it's hard to actually test if `to_owned` works or not.

Reply:
Your method is not unsafe and stable, isn't it?

Reply:
Fixed! I was actually able to test this more reliably by running `keypair` outside the thread with tiny stack.

> Your method is not unsafe and is stable, isn't it?

True. But it's incredibly ugly and not optimal. It involves `Vec -> Box_of_slice -> try convert to box of array`. It has an unwrap that should not need to be there.

Reply:
I don't think it's too much to ask for having a safe and clean way to "allocate byte buffer on the heap" in the standard library of a systems programming language :)

Reply:
> Maybe centralizing the util:alloc_boxed_array into a KeyBuffer::new_owned() method is also better, since you need to use this multiple times below.

I quickly tried this. But it did not make anything better. It required another method, and did not save much code in other places. So it was just more boilerplate than this. I think the current solution right now is good.